lib/Emitter.php

```diff
@@ -13,7 +13,7 @@
     private $iterator;
 
     public function __construct() {
-        $this->iterator = new class implements Iterator {
+        $this->iterator = new class {
```

Comment:
Just curious on your thought-process behind removing the `implements` expression here (for my own edification)?

Reply:
If you look at the new definition of `Internal\Producer`, I would need to add `advance as public; getCurrent as public` for this to still implement `Iterator`.